content/3-lifecycle/2-on-unmount/angularRenaissance/time.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, OnDestroy, signal } from "@angular/core";
+
+/* FIXME with rxjs + toSignal (but no need for onDestroy anymore)
+```ts
+@Component({
+  standalone: true,
+  selector: "app-time",
+  template: `<p>Current time: {{ time() }}</p>`,
+})
+export class TimeComponent {
+  time = toSignal(timer(0, 1000).pipe(
+    map(() => new Date().toLocaleTimeString()),
+  ));
+}
+```
+*/
+@Component({
+  standalone: true,
+  selector: "app-time",
+  template: `<p>Current time: {{ time() }}</p>`,
+})
+export class TimeComponent implements OnDestroy {
+  time = signal(new Date().toLocaleTimeString());
+
+  timer = setInterval(
```

Comment:
That's what I wrote above in the comment.
While I completely agree with you, I'm torn appart since the angular team is trying to make rxjs optional.
We both know that rxjs is the best fit here but if I were a non angular dev and wanted to take a look at how it is now, I think that rxjs could make me think that angular is still as hard as it was before... :confused:

Reply:
There is nothing hard here and `rxjs` compliments `signals` in a lot of ways. That's the right tool for the job here.
Also, Angular is still quite dependent on `rxjs` - the old http client and reactive forms are still here to stay for quite a while.

Reply:
You obviously are pretty good with rxjs, so am I. Since that's the case, you should know that many people struggle with it and use it the wrong way.
Since in the other frameworks/libs they all use their signal like solutions, as the goal is to compare frameworks/libs and not rxjs, while I would also use rxjs here too, in this context, it is pretty legit to do it with signals.
Since @matschik is the owner, he could give us his thought on this.

Reply:
Maybe it's just a bad example and instead of input focus it should be sth else?

Reply:
And yes, angular is still dependant on rxjs, but if you follow closely what they are doing, you have probably seen that it's a goal of theirs.
- They are making an interop package to be able to use rxjs, only when you need it
- In v19, they are adding a `linkedSignal` to help you use signals even more, to be able to sort of "internally update" the value inside a readonly signal. They are also adding `resource`, to be able to make http requests without the need of rxjs, and they are adding an `rxResource` in their interrop package for those who'll keep using rxjs when they need it.
- Nothing has been made yet or even said publicly, but they indeed have in mind to make the http client "rxjs free" and once again add some interrop tools (while I like it how it is, an rxjs based http client is not absolutely necessary. All you need is to wrap http requests in a place where angular can execute your interceptors, which can be made with a basic promise based client. If you still need an observable, a promise is an ObservableInput, so you can transform it the way you want). About forms, in general (since template driven form are also reactive, just as "reactive" forms), they can also make them rxjs free but there is even fewer information about it. Though, their goal remains the same :slightly_smiling_face:

Reply:
@LcsGa I'm more a React, Vue, Svelte developer but I can try.
From my research, using signals for reactive state management aligns closely with Angular’s current direction to reduce rxjs dependency. 

Also the original purpose of the snippet in Component Party is a minimal example on how to do unmount operations. Of course in this case, it's maybe not the best way to destroy a timer but it's explicit enough for more complex operations.